test-data/unit/check-possibly-undefined.test

```diff
@@ -1043,3 +1043,40 @@ def foo(x: Union[int, str]) -> None:
         assert_never(x)
     f  # OK
 [builtins fixtures/tuple.pyi]
+
+[case testForLoopBreakElseDefinition]
+# flags: --enable-error-code possibly-undefined
+# Regression test for https://github.com/python/mypy/issues/14209
+for i in [1, 2, 3]:
+    if i:
+        b = i
+        break
+else:
+    b = 1
+
+b  # OK
+
+[case testForLoopBreakElseException]
+# flags: --enable-error-code possibly-undefined
+# Regression test for https://github.com/python/mypy/issues/14209
+for i in [1, 2, 3]:
+    if i:
+        b = i
+        break
+else:
+    raise BaseException()
+
+b  # OK
+[builtins fixtures/exception.pyi]
```

Comment:
I am struggling with builtins/fixtures here.

Requirements:
- This test case needs a `for` loop with a definition and `break` in a conditional block, and a `raise` in the `else` block.
- Until #14209 is fixed, `possibly-undefined` is the only expected error.

The direct issue I have:
- With the default `builtins` (`lib-stub.builtins.pyi`), `list` is iterable, but there are no exceptions.
- With `fixtures/exception.pyi`, there is no iterable class, and `int` is not comparable`.
- I did not find a way to make this test case work.

What I tried:
- Without any fixture, I get a crash because `BaseException` is not defined (it is not defined in `lib-stub/builtins.pyi`).
- With `[builtins fixtures/exception.pyi]`, I get the expected hit for `possibly-undefined`, but also the false-positive `"List[int]" has no attribute "__iter__" (not iterable)` (`list` is only minimally defined in `fixtures/exception.pyi`).
- With an additional `[builtins fixture/list.pyi]`, i get `Duplicated filename tmp/builtins.pyi` (which makes sense).
- I also tried adding `class BaseException` via `[file builtins.py]`, which led to `Cannot find builtins.dict to add __annotations__`
- I tried putting many combinations of the content of `lib-stub/builtins.pyi`, `fixtures/exception.pyi`, `fixtures/int.pyi` and `fixtures/list.pyi` into `[file builtins.py]`, but always got either an error or more hits than just `possibly-undefined`.

Reply:
update `unit/fixtures/exception.pyi` to:
```py
class list:
    def __iter__(self): ...
```

Reply:
Thank you, I added a custom iterable class to not add additional runtime to all tests using `fixtures/exception.pyi`.